backend/src/main/kotlin/com/hiper2d/auth/AuthenticationTokenConverter.kt

```diff
@@ -0,0 +1,44 @@
+package com.hiper2d.auth
+
+import com.hiper2d.auth.token.GuestAuthenticationToken
+import com.hiper2d.auth.token.JwtPreAuthenticationToken
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureException
+import org.springframework.http.server.reactive.ServerHttpRequest
+import org.springframework.security.authentication.BadCredentialsException
+import org.springframework.security.core.Authentication
+import org.springframework.security.web.server.authentication.ServerAuthenticationConverter
+import org.springframework.web.server.ServerWebExchange
+import reactor.core.publisher.Mono
+
+class AuthenticationTokenConverter(private val jwtConfig: JwtConfigService): ServerAuthenticationConverter {
+
+    override fun convert(exchange: ServerWebExchange): Mono<Authentication> =
+        Mono.just(exchange.request)
+            .flatMap(this::extractJwtToken)
+            .switchIfEmpty(Mono.just(GuestAuthenticationToken()))
+
+    private fun extractJwtToken(request: ServerHttpRequest): Mono<Authentication> {
```

Comment:
val someshit = Mono.justOrEmpty(request.getTokenHeader())
            .filter { h -> h!!.isToken() }
            .map { h -> verifyAndBuildTokenObject(h!!.getToken()) }

Reply:
also instead of two separate istoken and gettoken it's better to make getToken return option and use justOrEmpty

Reply:
private fun extractJwtToken(request: ServerHttpRequest): Mono<Authentication> = 
        Mono.justOrEmpty(request.getTokenHeader())
            .handle<String>{ h, sink ->
                if (h != null && h.startsWith(jwtConfig.bearerPrefix))
                    sink.next(h.substring(jwtConfig.bearerPrefix.length + 1))
            }
            .flatMap { t -> verifyAndBuildTokenObject(t)}